gnat2goto/driver/arrays.ads

```diff
@@ -26,25 +30,29 @@ package Arrays is
    function Do_Array_Length (N : Node_Id) return Irep
      with Pre => Nkind (N) = N_Attribute_Reference;
 
-   function Do_Array_Range (N : Node_Id) return Irep
-     with Pre  => Nkind (N) = N_Range;
-
-   function Do_RHS_Array_Assign (N : Node_Id) return Irep_Array
-     with Pre => Nkind (N) in N_Op_Concat | N_Slice | N_Function_Call;
-
    function Get_Array_Component_Type (N : Node_Id) return Entity_Id
      with Post => Is_Type (Get_Array_Component_Type'Result);
 
+   function Do_Slice (N : Node_Id) return Irep
+     with Pre  => Nkind (N) = N_Slice;
+
+   function Do_Indexed_Component (N : Node_Id) return Irep
+     with Pre  => Nkind (N) = N_Indexed_Component;
+
+private
```

Comment:
If these functions are 'private', there's technically no need to mention them here in the `.ads` at all, unless you need them in a precondition/postcondition or something like that

Reply:
Intentionally there as a form of documentation.